Add `go vet`

Tests that are run in travis-ci for every PR should include `go vet` to check for suspicious constructs and formatting errors.

we can create `make test` target that will run all tests (vet,unit,cmd)

current state:

```
$ for f in `find . -path ./vendor -prune -o -name '*.go' -print `; do go vet $f; done
cli/app/app_test.go:109: missing argument for Errorf("%s"): format reads arg 3, have only 2 args
cli/app/app_test.go:112: missing argument for Errorf("%s"): format reads arg 3, have only 2 args
cli/app/app_test.go:115: missing argument for Errorf("%s"): format reads arg 3, have only 2 args
cli/app/app_test.go:118: missing argument for Errorf("%s"): format reads arg 3, have only 2 args
pkg/loader/bundle/bundle.go:109: no formatting directive in Fatalf call
pkg/loader/bundle/bundle.go:114: no formatting directive in Fatalf call
```

> We can also use tools like Go Meta Linter

The regular `golint` tool already finds a lot of issues. Might want to tackle that first.

```
$ for pkg in $( go list github.com/kubernetes-incubator/kompose/... | grep -v '/vendor/' ); do golint $pkg; done
/home/eric/src/github.com/kubernetes-incubator/kompose/cli/app/app.go:42:2: exported const DefaultComposeFile should have comment (or a comment on this block) or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/cli/command/command.go:67:1: comment on exported function ConvertCommandDummy should be of the form "ConvertCommandDummy ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/kobject/kobject.go:108:6: exported type ConvertOptions should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/kobject/kobject.go:161:1: exported function CheckUnsupportedKey should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/loader.go:28:6: exported type Loader should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/loader.go:42:15: should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/bundle/bundle.go:32:6: exported type Bundle should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/bundle/bundle.go:128:1: comment on exported method Bundle.LoadFile should be of the form "LoadFile ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/compose/compose.go:35:6: exported type Compose should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/loader/compose/compose.go:126:1: comment on exported method Compose.LoadFile should be of the form "LoadFile ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/transformer.go:24:6: exported type Transformer should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:31:1: comment on exported function CreateOutFile should be of the form "CreateOutFile ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:44:1: comment on exported function ParseVolume should be of the form "ParseVolume ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:80:1: comment on exported function ConfigLabels should be of the form "ConfigLabels ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:85:1: comment on exported function ConfigAnnotations should be of the form "ConfigAnnotations ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:95:1: comment on exported function TransformData should be of the form "TransformData ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/utils.go:117:1: comment on exported function Print should be of the form "Print ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:230:1: exported method Kubernetes.PortsExist should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:234:9: if block ends with a return statement, so drop this else and outdent its block
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:239:1: comment on exported method Kubernetes.CreateService should be of the form "CreateService ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:269:1: comment on exported method Kubernetes.UpdateKubernetesObjects should be of the form "UpdateKubernetesObjects ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:334:1: comment on exported method Kubernetes.SortServicesFirst should be of the form "SortServicesFirst ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/k8sutils.go:366:1: exported method Kubernetes.VolumesFrom should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:31:2: a blank import should be only in a main or test package, or have a comment justifying it
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:35:2: a blank import should be only in a main or test package, or have a comment justifying it
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:45:6: exported type Kubernetes should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:50:1: comment on exported const TIMEOUT should be of the form "TIMEOUT ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:54:1: comment on exported method Kubernetes.InitRC should be of the form "InitRC ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:84:1: comment on exported method Kubernetes.InitSvc should be of the form "InitSvc ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:102:1: comment on exported method Kubernetes.InitD should be of the form "InitD ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:129:1: comment on exported method Kubernetes.InitDS should be of the form "InitDS ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:155:1: comment on exported method Kubernetes.CreatePVC should be of the form "CreatePVC ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:187:1: comment on exported method Kubernetes.ConfigPorts should be of the form "ConfigPorts ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:200:1: comment on exported method Kubernetes.ConfigServicePorts should be of the form "ConfigServicePorts ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:220:1: comment on exported method Kubernetes.ConfigVolumes should be of the form "ConfigVolumes ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:282:1: comment on exported method Kubernetes.ConfigEmptyVolumeSource should be of the form "ConfigEmptyVolumeSource ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:289:1: comment on exported method Kubernetes.ConfigPVCVolumeSource should be of the form "ConfigPVCVolumeSource ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:299:1: comment on exported method Kubernetes.ConfigEnvs should be of the form "ConfigEnvs ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:312:1: comment on exported method Kubernetes.CreateKubernetesObjects should be of the form "CreateKubernetesObjects ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:355:1: comment on exported method Kubernetes.UpdateController should be of the form "UpdateController ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:376:1: comment on exported method Kubernetes.Deploy should be of the form "Deploy ..."
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:422:1: exported method Kubernetes.Undeploy should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:449:11: if block ends with a return statement, so drop this else and outdent its block
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:462:11: if block ends with a return statement, so drop this else and outdent its block
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes.go:470:11: if block ends with a return statement, so drop this else and outdent its block
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes_test.go:75:15: don't use leading k in Go names; func parameter kEnvs should be envs
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes_test.go:79:9: don't use leading k in Go names; range var kEnv should be env
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes_test.go:93:17: don't use leading k in Go names; func parameter kPorts should be ports
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes/kubernetes_test.go:97:9: don't use leading k in Go names; range var kPort should be port
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/openshift/openshift.go:44:6: exported type OpenShift should have comment or be unexported
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/openshift/openshift.go:57:9: if block ends with a return statement, so drop this else and outdent its block
/home/eric/src/github.com/kubernetes-incubator/kompose/pkg/transformer/openshift/openshift.go:174:1: exported method OpenShift.Deploy should have comment or be unexported
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,10 +12,14 @@ install:
   - true
 
 script:
+  - make validate
+  - make test-unit
+  
+  # make test-cmd requires kompose binary
   - make binary
+  
   # $GOPATH/bin is in $PATH
   - mkdir -p $GOPATH/bin
   - cp kompose $GOPATH/bin/
 
-  - make test-unit
   - make test-cmd

--- a/Makefile
+++ b/Makefile
@@ -26,6 +26,14 @@ clean:
 
 test-unit:
 	./script/make.sh test-unit
-
 test-cmd:
 	./script/make.sh test-cmd
+
+validate: gofmt vet
+
+vet:
+	./script/make.sh validate-vet
+lint:
+	./script/make.sh validate-lint
+gofmt:
+	./script/make.sh validate-gofmt

--- a/cli/app/app.go
+++ b/cli/app/app.go
@@ -241,7 +241,7 @@ func getTransformer(opt kobject.ConvertOptions) transformer.Transformer {
 	} else {
 		// Create/Init new OpenShift object that is initialized with a newly
 		// created Kubernetes object. Openshift inherits from Kubernetes
-		t = &openshift.OpenShift{kubernetes.Kubernetes{Opt: opt}}
+		t = &openshift.OpenShift{Kubernetes: kubernetes.Kubernetes{Opt: opt}}
 	}
 	return t
 }

--- a/cli/app/app_test.go
+++ b/cli/app/app_test.go
@@ -106,16 +106,16 @@ func TestParseVolume(t *testing.T) {
 			t.Errorf("In test case %q, returned unexpected error %v", test.test, err)
 		}
 		if name != test.name {
-			t.Errorf("In test case %q, returned volume name %s, expected %s", name, test.name)
+			t.Errorf("In test case %q, returned volume name %s, expected %s", test.test, name, test.name)
 		}
 		if host != test.host {
-			t.Errorf("In test case %q, returned host path %s, expected %s", host, test.host)
+			t.Errorf("In test case %q, returned host path %s, expected %s", test.test, host, test.host)
 		}
 		if container != test.container {
-			t.Errorf("In test case %q, returned container path %s, expected %s", container, test.container)
+			t.Errorf("In test case %q, returned container path %s, expected %s", test.test, container, test.container)
 		}
 		if mode != test.mode {
-			t.Errorf("In test case %q, returned access mode %s, expected %s", mode, test.mode)
+			t.Errorf("In test case %q, returned access mode %s, expected %s", test.test, mode, test.mode)
 		}
 	}
 }

--- a/cli/command/command.go
+++ b/cli/command/command.go
@@ -87,7 +87,7 @@ func generateBashCompletion(args []string) {
 	commonArgs := []string{"bundle", "file", "suppress-warnings", "verbose", "error-on-warning", "provider"}
 	flags := append(commonArgs, args...)
 
-	for _,f := range flags {
+	for _, f := range flags {
 		fmt.Printf("--%s\n", f)
 	}
 }

--- a/pkg/loader/bundle/bundle.go
+++ b/pkg/loader/bundle/bundle.go
@@ -133,12 +133,12 @@ func (b *Bundle) LoadFile(file string) kobject.KomposeObject {
 
 	buf, err := ioutil.ReadFile(file)
 	if err != nil {
-		logrus.Fatalf("Failed to read bundles file: ", err)
+		logrus.Fatalf("Failed to read bundles file: %s ", err)
 	}
 	reader := strings.NewReader(string(buf))
 	bundle, err := loadFile(reader)
 	if err != nil {
-		logrus.Fatalf("Failed to parse bundles file: ", err)
+		logrus.Fatalf("Failed to parse bundles file: %s", err)
 	}
 
 	for name, service := range bundle.Services {

--- a/pkg/transformer/kubernetes/k8sutils.go
+++ b/pkg/transformer/kubernetes/k8sutils.go
@@ -245,6 +245,7 @@ func (k *Kubernetes) CreateService(name string, service kobject.ServiceConfig, o
 	svc.Spec.Ports = servicePorts
 
 	svc.Spec.Type = api.ServiceType(service.ServiceType)
+
 	// Configure annotations
 	annotations := transformer.ConfigAnnotations(service)
 	svc.ObjectMeta.Annotations = annotations

--- a/pkg/transformer/kubernetes/kubernetes.go
+++ b/pkg/transformer/kubernetes/kubernetes.go
@@ -382,7 +382,7 @@ func (k *Kubernetes) Deploy(komposeObject kobject.KomposeObject, opt kobject.Con
 	if !opt.EmptyVols {
 		pvcStr = " and PersistentVolumeClaims "
 	}
-	fmt.Println("We are going to create Kubernetes Deployments, Services" +pvcStr+ "for your Dockerized application. \n" +
+	fmt.Println("We are going to create Kubernetes Deployments, Services" + pvcStr + "for your Dockerized application. \n" +
 		"If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead. \n")
 
 	factory := cmdutil.NewFactory(nil)
@@ -424,7 +424,7 @@ func (k *Kubernetes) Deploy(komposeObject kobject.KomposeObject, opt kobject.Con
 	} else {
 		pvcStr = ""
 	}
-	fmt.Println("\nYour application has been deployed to Kubernetes. You can run 'kubectl get deployment,svc,pods" +pvcStr+ "' for details.")
+	fmt.Println("\nYour application has been deployed to Kubernetes. You can run 'kubectl get deployment,svc,pods" + pvcStr + "' for details.")
 
 	return nil
 }

--- a/pkg/transformer/openshift/openshift.go
+++ b/pkg/transformer/openshift/openshift.go
@@ -178,7 +178,7 @@ func (o *OpenShift) Deploy(komposeObject kobject.KomposeObject, opt kobject.Conv
 	if !opt.EmptyVols {
 		pvcStr = " and PersistentVolumeClaims "
 	}
-	fmt.Println("We are going to create OpenShift DeploymentConfigs, Services" +pvcStr+ "for your Dockerized application. \n" +
+	fmt.Println("We are going to create OpenShift DeploymentConfigs, Services" + pvcStr + "for your Dockerized application. \n" +
 		"If you need different kind of resources, use the 'kompose convert' and 'oc create -f' commands instead. \n")
 
 	// initialize OpenShift Client
@@ -238,7 +238,7 @@ func (o *OpenShift) Deploy(komposeObject kobject.KomposeObject, opt kobject.Conv
 	} else {
 		pvcStr = ""
 	}
-	fmt.Println("\nYour application has been deployed to OpenShift. You can run 'oc get dc,svc,is" +pvcStr+ "' for details.")
+	fmt.Println("\nYour application has been deployed to OpenShift. You can run 'oc get dc,svc,is" + pvcStr + "' for details.")
 
 	return nil
 }

--- a/script/.validate
+++ b/script/.validate
@@ -14,34 +14,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ -z "$VALIDATE_UPSTREAM" ]; then
-	# this is kind of an expensive check, so let's not do this twice if we
-	# are running more than one validate bundlescript
-	
-	VALIDATE_REPO='https://github.com/kubernetes-incubator/kompose.git'
-	VALIDATE_BRANCH='master'
-	
-	if [ "$TRAVIS" = 'true' -a "$TRAVIS_PULL_REQUEST" != 'false' ]; then
-		VALIDATE_REPO="https://github.com/${TRAVIS_REPO_SLUG}.git"
-		VALIDATE_BRANCH="${TRAVIS_BRANCH}"
-	fi
-	
-	VALIDATE_HEAD="$(git rev-parse --verify HEAD)"
-	
-	git fetch -q "$VALIDATE_REPO" "refs/heads/$VALIDATE_BRANCH"
-	VALIDATE_UPSTREAM="$(git rev-parse --verify FETCH_HEAD)"
-	
-	VALIDATE_COMMIT_LOG="$VALIDATE_UPSTREAM..$VALIDATE_HEAD"
-	VALIDATE_COMMIT_DIFF="$VALIDATE_UPSTREAM...$VALIDATE_HEAD"
-	
-	validate_diff() {
-		if [ "$VALIDATE_UPSTREAM" != "$VALIDATE_HEAD" ]; then
-			git diff "$VALIDATE_COMMIT_DIFF" "$@"
-		fi
-	}
-	validate_log() {
-		if [ "$VALIDATE_UPSTREAM" != "$VALIDATE_HEAD" ]; then
-			git log "$VALIDATE_COMMIT_LOG" "$@"
-		fi
-	}
-fi
+# all Kompose packages (skip vendor dir)
+PKGS=$( go list -e ./... | grep -v '/vendor/' )

--- a/script/make.sh
+++ b/script/make.sh
@@ -25,10 +25,6 @@ DEFAULT_BUNDLES=(
 	validate-lint
 	validate-vet
 	binary
-
-	# test-unit
-	# test-integration
-	# test-acceptance
 )
 bundle() {
     local bundle="$1"; shift

--- a/script/validate-gofmt
+++ b/script/validate-gofmt
@@ -16,29 +16,24 @@
 
 source "$(dirname "$BASH_SOURCE")/.validate"
 
-IFS=$'\n'
-files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/' || true) )
-unset IFS
+GO_FILES=$(find . -path ./vendor -prune -o -name '*.go' -print )
 
-badFiles=()
-for f in "${files[@]}"; do
-	# we use "git show" here to validate that what's committed is formatted
-	if [ "$(git show "$VALIDATE_HEAD:$f" | gofmt -s -l)" ]; then
-		badFiles+=( "$f" )
+
+for file in $GO_FILES; do
+	gofmtOutput=$(gofmt -l "$file")
+	if [ "$gofmtOutput" ]; then
+		errors+=("$gofmtOutput")
 	fi
-done
+done 
 
-if [ ${#badFiles[@]} -eq 0 ]; then
-	echo 'Congratulations!  All Go source files are properly formatted.'
+
+
+if [ ${#errors[@]} -eq 0 ]; then
+	echo "gofmt OK"
 else
-	{
-		echo "These files are not properly gofmt'd:"
-		for f in "${badFiles[@]}"; do
-			echo " - $f"
-		done
-		echo
-		echo 'Please reformat the above files using "gofmt -s -w" and commit the result.'
-		echo
-	} >&2
-	false
+	echo "gofmt ERROR - These files are not formated by gofmt:"
+	for err in "${errors[@]}"; do
+		echo "$err"
+	done
+	exit 1
 fi

--- a/script/validate-lint
+++ b/script/validate-lint
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
 
 # Copyright 2016 The Kubernetes Authors All rights reserved.
 #
@@ -16,33 +17,21 @@
 
 source "$(dirname "$BASH_SOURCE")/.validate"
 
-# We will eventually get to the point where packages should be the complete list
-# of subpackages, vendoring excluded, as given by:
-#
-IFS=$'\n'
-files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/\|^integration' || true) )
-unset IFS
-
-errors=()
-for f in "${files[@]}"; do
-	# we use "git show" here to validate that what's committed passes go lint
-	failedLint=$(golint "$f")
-	if [ "$failedLint" ]; then
-		errors+=( "$failedLint" )
+for pkg in $PKGS; do
+	lintOutput=$(golint "$pkg")
+	# if lineOutput is not empty, save it to errros array
+	if [ "$lintOutput" ]; then
+		errors+=("$lintOutput")
 	fi
-done
+done 
+
 
 if [ ${#errors[@]} -eq 0 ]; then
-	echo 'Congratulations!  All Go source files have been linted.'
+	echo "golint OK"
 else
-	{
-		echo "Errors from golint:"
-		for err in "${errors[@]}"; do
-			echo "$err"
-		done
-		echo
-		echo 'Please fix the above errors. You can test via "golint" and commit the result.'
-		echo
-	} >&2
-	false
+	echo "golint ERRORS:"
+	for err in "${errors[@]}"; do
+		echo "$err"
+	done
+	exit 1
 fi

--- a/script/validate-vet
+++ b/script/validate-vet
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
 
 # Copyright 2016 The Kubernetes Authors All rights reserved.
 #
@@ -16,31 +17,5 @@
 
 source "$(dirname "$BASH_SOURCE")/.validate"
 
-IFS=$'\n'
-files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/' || true) )
-unset IFS
-
-errors=()
-for f in "${files[@]}"; do
-	# we use "git show" here to validate that what's committed passes go vet
-	failedVet=$(go vet "$f")
-	if [ "$failedVet" ]; then
-		errors+=( "$failedVet" )
-	fi
-done
-
-
-if [ ${#errors[@]} -eq 0 ]; then
-	echo 'Congratulations!  All Go source files have been vetted.'
-else
-	{
-		echo "Errors from go vet:"
-		for err in "${errors[@]}"; do
-			echo " - $err"
-		done
-		echo
-		echo 'Please fix the above errors. You can test via "go vet" and commit the result.'
-		echo
-	} >&2
-	false
-fi
+go vet $PKGS;
+ 

